src/main/java/com/newrelic/experts/jenkins/extensions/EventCollectorWork.java

```diff
@@ -77,49 +77,59 @@ public long getRecurrencePeriod() {
   
   @Override
   protected void doRun() throws Exception {
-    final String methodName = "doRun";
-    final boolean isLoggingTrace = LOGGER.isLoggable(Level.FINE);
-    final boolean isLoggingDebug = LOGGER.isLoggable(Level.FINEST);
+    synchronized (this.recorder) {
+      final String methodName = "doRun";
+      final boolean isLoggingTrace = LOGGER.isLoggable(Level.FINE);
+      final boolean isLoggingDebug = LOGGER.isLoggable(Level.FINEST);
 
-    if (isLoggingTrace) {
-      LOGGER.entering(CLASS_NAME, methodName);
-    }
+      if (isLoggingTrace)
+      {
+        LOGGER.entering(CLASS_NAME, methodName);
+      }
 
-    Event[] events = this.recorder.popEvents();
+      Event[] events = this.recorder.popEvents();
 
-    if (events.length == 0) {
-      if (isLoggingDebug) {
-        LOGGER.logp(Level.FINE, CLASS_NAME, methodName,
-            "RETURN EARLY No events to send."
-        );
+      if (events.length == 0)
+      {
+        if (isLoggingDebug)
+        {
+          LOGGER.logp(Level.FINE, CLASS_NAME, methodName,
+              "RETURN EARLY No events to send."
+          );
+        }
+        return;
       }
-      return;
-    }
-    
-    StandardUsernamePasswordCredentials credentialsId =
-        this.nrjConfig.getInsightsInsertCredentials();
-    if (credentialsId == null) {
-      if (isLoggingTrace) {
-        LOGGER.logp(Level.FINE, CLASS_NAME, methodName,
-            "RETURN EARLY No credentials."
+
+      StandardUsernamePasswordCredentials credentialsId =
+          this.nrjConfig.getInsightsInsertCredentials();
+      if (credentialsId == null)
+      {
+        if (isLoggingTrace)
+        {
+          LOGGER.logp(Level.FINE, CLASS_NAME, methodName,
+              "RETURN EARLY No credentials."
+          );
+        }
+        return;
+      }
+
+      try
+      {
+        this.client.recordEvents(
+            JenkinsUtils.createClientConnectionConfig(this.client),
+            credentialsId.getUsername(),
+            Secret.toString(credentialsId.getPassword()),
+            events
         );
+      } catch (NewRelicClientException nrce)
+      {
+        LOGGER.log(Level.SEVERE, "Failed to post events");
       }
-      return;
-    }
 
-    try {
-      this.client.recordEvents(
-          JenkinsUtils.createClientConnectionConfig(this.client),
-          credentialsId.getUsername(),
-          Secret.toString(credentialsId.getPassword()),
-          events
-      );
-    } catch (NewRelicClientException nrce) {
-      LOGGER.log(Level.SEVERE, "Failed to post events");
-    }
-    
-    if (isLoggingTrace) {
-      LOGGER.exiting(CLASS_NAME, methodName);
+      if (isLoggingTrace)
+      {
+        LOGGER.exiting(CLASS_NAME, methodName);
+      }
```

Comment:
@dalvizu All the formatting changes above should go away once we move the `synchronized` block down into the recorder.  Also, please abide by the `checkstyle` rules.  Opening braces should go on the same line as the `if`.